### PR TITLE
Auction sample clean up

### PR DESCRIPTION
**Pull request by nikhil550**
Some code cleanup for the auction sample

- Use base64 decoded identities
- use correct variable names
- re-use internal smart contract functions
- application code cleanup

Signed-off-by: Nikhil Gupta <ngupta@symbridge.com>

### CODE REVIEW

**Review by lehors — CHANGES_REQUESTED** [review]

**Comment by lehors** [comment]
Looks good overall but when I went through it I noticed that the output when trying to end the auction early is no longer the same. Here is what I got:

```$ node endAuction.js org1 seller PaintingAuction
Loaded the network configuration located at /Users/lehors/Projects/github.com/hyperledger/fabric-samples/test-network/organizations/peerOrganizations/org1.example.com/connection-org1.json
Built a file system wallet at /Users/lehors/Projects/github.com/hyperledger/fabric-samples/auction/application-javascript/wallet/org1

--> Submit the transaction to end the auction
2021-01-26T09:43:59.821Z - error: [DiscoveryHandler]: compareProposalResponseResults[undefined] - read/writes result sets do not match index=1
2021-01-26T09:43:59.824Z - error: [Transaction]: Error: No valid responses from any peers. Errors:
    peer=undefined, status=grpc, message=Peer endorsements do not match
******** FAILED to submit bid: Error: No valid responses from any peers. Errors:
    peer=undefined, status=grpc, message=Peer endorsements do not match
```

**Comment by nikhil550** [comment]
> Looks good overall but when I went through it I noticed that the output when trying to end the auction early is no longer the same. Here is what I got:
> 
> ```
> Loaded the network configuration located at /Users/lehors/Projects/github.com/hyperledger/fabric-samples/test-network/organizations/peerOrganizations/org1.example.com/connection-org1.json
> Built a file system wallet at /Users/lehors/Projects/github.com/hyperledger/fabric-samples/auction/application-javascript/wallet/org1
> 
> --> Submit the transaction to end the auction
> 2021-01-26T09:43:59.821Z - error: [DiscoveryHandler]: compareProposalResponseResults[undefined] - read/writes result sets do not match index=1
> 2021-01-26T09:43:59.824Z - error: [Transaction]: Error: No valid responses from any peers. Errors:
>     peer=undefined, status=grpc, message=Peer endorsements do not match
> ******** FAILED to submit bid: Error: No valid responses from any peers. Errors:
>     peer=undefined, status=grpc, message=Peer endorsements do not match
> ```

Replaced with the new output

File: auction/chaincode-go/smart-contract/auction.go [hunk 1]
```diff
@@ -422,66 +389,54 @@ func (s *SmartContract) CloseAuction(ctx contractapi.TransactionContextInterface
...
-		return fmt.Errorf("Auction interest object %v not found", auctionID)
-	}
-
-	var auctionJSON Auction
-	err = json.Unmarshal(auctionBytes, &auctionJSON)
-	if err != nil {
-		return fmt.Errorf("failed to create auction object JSON: %v", err)
-	}
+	// get auction from public state
+	auction, err := s.QueryAuction(ctx,auctionID)
```
[the remark this answers is not in the document]
Comment (nikhil550) [reply auction/chaincode-go/smart-contract/auction.go:402]: done.

File: auction/chaincode-go/smart-contract/auction.go [hunk 2]
```diff
@@ -373,44 +352,32 @@ func (s *SmartContract) RevealBid(ctx contractapi.TransactionContextInterface, a
...
```
[the remark this answers is not in the document]
Comment (nikhil550) [reply auction/chaincode-go/smart-contract/auction.go:362]: Done.

File: auction/chaincode-go/smart-contract/auction.go [hunk 3]
```diff
@@ -156,20 +156,11 @@ func (s *SmartContract) SubmitBid(ctx contractapi.TransactionContextInterface, a
...
-
-	if auctionBytes == nil {
-		return fmt.Errorf("Auction not found: %v", auctionID)
-	}
-	err = json.Unmarshal(auctionBytes, &auctionJSON)
-	if err != nil {
-		return fmt.Errorf("failed to create auction object JSON: %v", err)
-	}
+	// get the auction from public state
+	auction, err := s.QueryAuction(ctx,auctionID)
```
[the remark this answers is not in the document]
Comment (nikhil550) [reply auction/chaincode-go/smart-contract/auction.go:160]: done.

File: auction/README.md [hunk 4]
```diff
@@ -328,8 +319,11 @@ The output should look something like the following:
 
 ```
 --> Submit the transaction to end the auction
-2020-11-06T13:16:11.591Z - warn: [TransactionEventHandler]: strategyFail: commit failure for transaction "99feade5b7ec223839200867b57d18971c3e9f923efc95aaeec720727f927366": TransactionError: Commit of transaction 99feade5b7ec223839200867b57d18971c3e9f923efc95aaeec720727f927366 failed on peer peer0.org1.example.com:7051 with status ENDORSEMENT_POLICY_FAILURE
-******** FAILED to submit bid: TransactionError: Commit of transaction 99feade5b7ec223839200867b57d18971c3e9f923efc95aaeec720727f927366 failed on peer peer0.org1.example.com:7051 with status ENDORSEMENT_POLICY_FAILURE
+2021-01-28T16:47:27.501Z - error: [DiscoveryHandler]: compareProposalResponseResults[undefined] - read/writes result sets do not match index=1
+2021-01-28T16:47:27.503Z - error: [Transaction]: Error: No valid responses from any peers. Errors:
+    peer=undefined, status=grpc, message=Peer endorsements do not match
+******** FAILED to submit bid: Error: No valid responses from any peers. Errors:
+    peer=undefined, status=grpc, message=Peer endorsements do not match
```
Comment (lehors) [inline auction/README.md:326]: Does anyone know why instead of the explicit error we used to get we now get this rather cryptic one? @denyeart ?

Comment (nikhil550) [reply auction/README.md:326]: The error actually reflects what is happening in the sample. one organization is trying to close the auction, while another organization is rejecting the transaction due to a non disclosed bid. As a result, the endorsements from the two peers do not match.

**Comment by lindluni** [comment]
Currently an outage at Google Cloud where our Artifactory instance is hosted, which is causing the failure here as the Maven repo is timing out trying to fetch the artifact, will retry later when the outage is resolved:

https://status.jfrog.io/

**Review by lehors — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**